pkg/metrics/backends/prometheus/prometheus.go

```diff
@@ -93,3 +97,51 @@ func (b *Backend) QueryInstant(ctx context.Context, query string, ts time.Time)
 		return nil, errors.Errorf("unknown supported type: '%q'", v)
 	}
 }
+
+// getAlertmanagers returns Alertmanager clients that is associated with the backend.
+func (b *Backend) getAlertmanagers(ctx context.Context) ([]*amclient.Alertmanager, error) {
```

Comment:
It's quite wasteful to discover Alertmanagers every single healer queries, is there any ways we can cache or in-mem stored those AMs? It does not change frequently.

Reply:
I used a separate ticker loop in the first place.